crates/api_crud/src/post/create.rs

```diff
@@ -92,25 +92,15 @@ impl PerformCrud for CreatePost {
     let (embed_title, embed_description, embed_video_url) = metadata_res
       .map(|u| (u.title, u.description, u.embed_video_url))
       .unwrap_or_default();
-    let language = data.language.clone();
-    let mut language = blocking(context.pool(), move |conn| {
-      Language::read_id_from_code_opt(conn, language)
-    })
-    .await??;
-    // if user only speaks one language, use that as post language. otherwise, set it as "undetermined"
-    if language.is_none() {
-      let user_langs = local_user_view.local_user.discussion_languages;
-      language = if user_langs.len() == 1 {
-        Some(user_langs[0])
-      } else {
-        Some(
-          blocking(context.pool(), move |conn| {
-            Language::read_undetermined(conn)
-          })
-          .await??,
-        )
-      };
-    };
+
+    let language_id = Some(
+      data.language_id.unwrap_or(
+        blocking(context.pool(), move |conn| {
+          Language::read_undetermined(conn)
+        })
+        .await??,
+      ),
+    );
```

Comment:
Not sure about this, might be better to throw an error. Otherwise, a couple of lazy users can make the entire feature unusable.

Reply:
If its optional, we should still let it work if its not provided. What's most important is that the main clients, like lemmy-ui, are going to provide the language, which of course we will.